packages/components/src/toolbar/stories/index.tsx

```diff
@@ -82,34 +82,31 @@ Default.args = {
 			</ToolbarGroup>
 			<ToolbarGroup>
 				<ToolbarItem>
-					{
-						// @ts-expect-error TODO: Remove when DropdownMenu is typed
-						( toggleProps ) => {
-							return (
-								<DropdownMenu
-									hasArrowIndicator
-									icon={ alignLeft }
-									label="Align"
-									controls={ [
-										{
-											icon: alignLeft,
-											title: 'Align left',
-											isActive: true,
-										},
-										{
-											icon: alignCenter,
-											title: 'Align center',
-										},
-										{
-											icon: alignRight,
-											title: 'Align right',
-										},
-									] }
-									toggleProps={ toggleProps }
-								/>
-							);
-						}
-					}
+					{ /* There is an issue here with TS not recognizing the
+					 * `RenderProp` being passed.
+					 * @ts-expect-error */ }
+					{ ( toggleProps ) => (
```

Comment:
Hey @diegohaz 👋  We're refactoring this component to TypeScript, and we're getting an error around the type of `children`. We're passing a render prop, but for some reason it doesn't seem to work with the type that `ToolbarItem` from `reakit` is expecting. Do you have any ideas why this is happening?

Reply:
What's the error? You might need to explicitly type the parameter (instead of relying on inference).

Reply:
This is the error:

```
Type '(toggleProps: ExtractHTMLAttributes<any>) => Element' is not assignable to type '((string | number | boolean | ReactElement<any, string | JSXElementConstructor<any>> | ReactFragment | ReactPortal) & (string | ... 5 more ... | RenderProp<...>)) | null | undefined'.
  Type '(toggleProps: ExtractHTMLAttributes<any>) => Element' is not assignable to type 'string & RenderProp<ExtractHTMLAttributes<any>>'.
    Type '(toggleProps: ExtractHTMLAttributes<any>) => Element' is not assignable to type 'string'.ts(2322)
```

What I don't understand is the initial `string & ...` in the expected type. Do you have any ideas as to why?

Reply:
(also, found this related comment about `ToolbarItem`'s children types https://github.com/WordPress/gutenberg/pull/50262#issuecomment-1531967931 )

Reply:
Hey @diegohaz , did you have time to take a look at this?

Reply:
Looks like it's coming from https://github.com/ariakit/ariakit/blob/3966597f6a24f601f83d0816231486b8a293faee/packages/reakit-utils/src/types.ts#L63

But I also don't understand why there's a `string & ...` there. It should be possible to fix this in the `ToolbarItem` component by providing own types for the `children` prop instead of inheriting them directly from Reakit:

https://github.com/WordPress/gutenberg/blob/76319f8ac3a68c01966e6ba745511bf5a4677756/packages/components/src/toolbar/toolbar-item/index.tsx#L23

Reply:
Awesome, thanks for that @diegohaz. I'll put that on my list of things to follow up on!

Reply:
Thank you, @diegohaz . I'll circle back to you in case we understand where the `string & ...` comes from in the resulting type